ample/util/tests/test_options_processor.py

```diff
@@ -82,7 +83,7 @@ def test_options_shelxe_resolution_cmd(self):
         
         options.d['use_shelxe'] = True
         options.d['shelxe_rebuild'] = True
-        options.d['mtz_min_resolution'] = SHELXE_MAX_PERMITTED_RESOLUTION_CC + (extra / 2)
+        options.d['mtz_min_resolution'] = random.uniform(limit, REBUILD_MAX_PERMITTED_RESOLUTION)
```

Comment:
You should *really* be using [hypothesis](https://hypothesis.readthedocs.io/en/latest/) for this test. This is non-deterministic, despite passing it really doesn't allow for proper reproducibility of the test.

Reply:
Ooh - that is interesting. Thanks for pointing that out. I don't have time to look into that now (and don't want to add another dependency), but that's definitely good to know about.

Reply:
Yes, it's really really useful, also as a state machine if you want to test particular actions / states of systems. The nicest bit is it stores previous failures / test cases and trials new situations plus ensures all failures work.